### PR TITLE
firedancer: tpool working with testnet

### PR DESCRIPTION
**Pull request by llamb-jump**
…ges because fdctl

does not yet support extra huge pages and the topo does not yet support the inclusion of anonymous workspaces. See:
  fd_topo_tile_extra_huge_pages
  fd_topo_huge_page_cnt

### CODE REVIEW

File: src/app/fdctl/run/tiles/fd_replay.c [hunk 1]
```diff
@@ -247,19 +248,19 @@ after_frag( void *             _ctx,
       FD_LOG_ERR(( "unexpected fork switch mid block execution, cannot continue" ));
     }
 
-    int res = fd_runtime_prepare_txns( &fork->slot_ctx, task_info, txns, txn_cnt );
-    if( res != 0 && !( ctx->flags & REPLAY_FLAG_PACKED_MICROBLOCK ) ) {
-      FD_LOG_WARNING(( "block invalid - slot: %lu", ctx->curr_slot ));
-      // TODO: need to yeet this block from the replay frontier.
-      return;
-    }
+    // int res = fd_runtime_prepare_txns( &fork->slot_ctx, task_info, txns, txn_cnt );
```
Comment (asiegel-jt) [inline src/app/fdctl/run/tiles/fd_replay.c:251]: Why is this commented out?

File: src/app/fdctl/run/tiles/fd_replay.c [hunk 2]
```diff
@@ -544,11 +554,21 @@ unprivileged_init( fd_topo_t *      topo,
   ctx->replay->acc_mgr      = ctx->slot_ctx->acc_mgr;
   ctx->replay->funk         = funk;
 
-  ctx->tpool = fd_tpool_init( ctx->tpool_mem, 1 );
-  ctx->max_workers = 1;
+  ctx->max_workers = tile->replay.tpool_thread_count;
```
Comment (asiegel-jt) [inline src/app/fdctl/run/tiles/fd_replay.c:557]: +1 to include the main thread, unless the thread count already does

File: src/app/fdctl/run/tiles/fd_replay.c [hunk 3]
```diff
@@ -544,11 +554,21 @@ unprivileged_init( fd_topo_t *      topo,
...
+  tpool_boot( (ushort)tile->cpu_idx, ctx->max_workers );
+  ctx->tpool = fd_tpool_init( ctx->tpool_mem, ctx->max_workers );
+
+  /* start the tpool workers */
+  ulong tpool_worker_stack_sz = (1UL<<23UL); /* 8MB */
```
Comment (asiegel-jt) [inline src/app/fdctl/run/tiles/fd_replay.c:562]: 256MB

File: src/app/fddev/main1.c [hunk 4]
```diff
@@ -81,6 +85,7 @@ fd_topo_run_tile_t * TILES[] = {
   &fd_tile_repair,
   &fd_tile_store_int,
   &fd_tile_replay,
+  &fd_tile_tvu_thread,
```
Comment (asiegel-jt) [inline src/app/fddev/main1.c:88]: tile_replay_thread?

File: src/app/fdctl/run/tiles/fd_replay.c (hunk 1 above)
Comment (llamb-jump) [reply src/app/fdctl/run/tiles/fd_replay.c:251]: it was replaced with `fd_runtime_execute_txns_in_waves_tpool`

File: src/app/fdctl/run/tiles/fd_replay.c (hunk 2 above)
Comment (llamb-jump) [reply src/app/fdctl/run/tiles/fd_replay.c:557]: it includes the main thread this way

File: src/app/fddev/main1.c (hunk 4 above)
Comment (llamb-jump) [reply src/app/fddev/main1.c:84]: updated

File: contrib/test/test_firedancer_testnet.sh [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+trap cleanup EXIT SIGINT SIGTERM
+sudo killall fddev || true
+
+# if fddev is not on path then use the one in the home directory
+if ! command -v fddev > /dev/null; then
+  PATH="$FD_DIR/build/native/gcc/bin":$PATH
+fi
+
+ENTRYPOINT=emam-ccn-solana-testnet-val1.jumpisolated.com
```
Comment (lheeger-jump) [inline contrib/test/test_firedancer_testnet.sh:45]: this should be the public entrypoint by default. people outside cannot use this script.

**Review by lheeger-jump — CHANGES_REQUESTED** [review]

**Review by asiegel-jt — DISMISSED** [review]

**Review by lheeger-jump — DISMISSED** [review]

**Review by ibhatt-jumptrading — APPROVED** [review]

**Pull request merged**